src/util/drawing.js

```diff
@@ -0,0 +1,175 @@
+import external from '../externalModules.js';
+import { toolStyle, textStyle } from '../index.js';
+
+export function getNewContext (canvas) {
+  // Return a new "HTML Canvas 2D Context"
+  // https://www.w3.org/TR/2dcontext/
+  const context = canvas.getContext('2d');
+
+  // Set the context transform to the identity matrix.
+  // https://www.w3.org/TR/2dcontext/#transformations
+  context.setTransform(1, 0, 0, 1, 0, 0);
+
+  return context;
+}
+
+export function draw (context, fn) {
+  // Perform canvas drawing operations on a new state-stack of context
+  // https://www.w3.org/TR/2dcontext/#the-canvas-state
+  context.save();
+  fn(context);
+  context.restore();
+}
+
+export function path (context, options, fn) {
+
+  const { color, lineWidth, fillStyle, lineDash } = options;
+
+  context.beginPath();
+  context.strokeStyle = color || context.strokeStyle;
+  context.lineWidth = lineWidth || (lineWidth === undefined && toolStyle.getToolWidth()) || context.lineWidth;
+  if (lineDash) {
+    context.setLineDash(lineDash);
+  }
+
+  fn(context);
+
+  if (fillStyle) {
+    context.fillStyle = fillStyle;
+    context.fill();
+  }
+  context.stroke();
+  if (lineDash) {
+    context.setLineDash([]);
+  }
+}
+
+export function setShadow (context, options) {
+  if (options.shadow) {
```

Comment:
This seems a bit weird. If I was using this function I would assume it would just be:

````javascript
options: {
    shadowColor: '#000',
    shadowOffsetX: 1,
    shadowOffsetY: 1
}
````

rather than a subobject with .shadow: { ... } 

Thoughts?

Reply:
Ah right you're including the whole if block. OK I guess it's fine